File: CPP/crud.cpp
#include <iostream>
#include <string>

using namespace std;

//deklarasi kelas crud
class Crud{
	
	private:
		vector<Mahasiswa> mhslist; //untuk list yang berisikan atribut mahasiswa (karena dia keturunan terakhir maka dia mewarisi semua atribut dati parent parentnya)
		vector<string> niklist; //untuk list yang berisikan hanya atribut nik nya saja sebagai primary key
	public:

		//constructor
		Crud(){

		}

		//prosedur untuk menampilkan data dalam list
		void show(){
			
			//jika list sizenya sama dengan 0 (tidak ada data)
			if(mhslist.size() == 0){
				cout << "\nThere is no data, please add data first\n";
			} //jika tidak sama dengan 0 (ada data)
			else{
				cout << "\nList of Student :\n";

				//print datanya sebanyak banyaknya data dalam list mahasiswa

				for(int i = 0; i < mhslist.size(); i++){
					cout<< "Student no  " << i+1 << "\n";
					cout<< "NIK 		: " << mhslist[i].get_nik() << '\n';
					cout<< "Name 		: " << mhslist[i].get_nama() << '\n';
					cout<< "Gender 		: " << mhslist[i].get_gender() << '\n';
					cout<< "University	: " << mhslist[i].get_univ() << '\n';
					cout<< "email 		: " << mhslist[i].get_email() << '\n';
					cout<< "NIM 		: " << mhslist[i].get_nim() << '\n';
					cout<< "Degree 		: " <<mhslist[i].get_prodi() <<'\n';
					cout<< "Faculty 	: " <<mhslist[i].get_fakultas() << '\n';

					cout << '\n';
				}
			}
		}

		//prosedur untuk menambahkan data
		void add(Mahasiswa temp){
			//menggunakan fungsi find untuk mencari data dengan nik yang sama pada list nik
			//jika hasilnya tidak sama dengan akhir dari niklist, berarti data ditemukan
			if(find(niklist.begin(), niklist.end(), temp.get_nik()) != niklist.end()){
				//karena nim yang ingin ditambahkan sudah ada pada data
				//maka tampilkan pesan nim sudah digunakan
				cout << "\nNIK is already used, please use another NIK\n";
			}
			else{ //jika tidak

				this->mhslist.push_back(temp); //masukkan semua atribut ke dalam list mahasiswa
				this->niklist.push_back(temp.get_nik()); //dan masukan nik saja ke dalam list nik
			}
		}

		//destructor
		~Crud(){

		}

};

File: Python/Crud.py
#deklarasi kelas crud
class Crud:
	
	__mhslist = [] #untuk list yang berisikan atribut mahasiswa (karena dia keturunan terakhir maka dia mewarisi semua atribut dati parent parentnya)
	__niklist = [] #untuk list yang berisikan hanya atribut nik nya saja sebagai primary key


	#prosedur untuk menampilkan data dalam list
	def show(self):
		if len(self.__mhslist) == 0:
			print("There is no data, please add data first");
		else:
			print ("List of Student :")

			#print datanya sebanyak data dalam list mahasiswa

			for i in range(len(self.__mhslist)):
				print ("Student no ", i+1)
				print ("NIK 		: " , self.__mhslist[i].get_nik())
				print ("Name 		: " , self.__mhslist[i].get_nama())
				print ("Gender 		: " , self.__mhslist[i].get_gender())
				print ("University 	: " , self.__mhslist[i].get_univ())
				print ("email 		: " , self.__mhslist[i].get_email())
				print ("NIM 		: " , self.__mhslist[i].get_nim())
				print ("Degree 		: " , self.__mhslist[i].get_prodi())
				print ("Faculty 	: " , self.__mhslist[i].get_fakultas(), "\n")

	#prosedur untuk menambahkan data
	def add(self, temp):
		#jika nik sudah ada pada data
		if(temp.get_nik() in self.__niklist):
			#karena nik yang ingin ditambahkan sudah ada pada data
			#maka tampilkan pesan nik sudah digunakan
			print("\nNIK is already used, please use another NIK")
		else: #jika tidak
			self.__mhslist.append(temp); #masukkan semua atribut ke dalam list mahasiswa
			self.__niklist.append(temp.get_nik()); #dan masukan nik saja ke dalam list nik